sympy/core/evalf.py

```diff
@@ -1750,3 +1753,56 @@ def N(x, n=15, **options):
     # by using rational=True, any evaluation of a string
     # will be done using exact values for the Floats
     return sympify(x, rational=True).evalf(n, **options)
+
+
+def evalf_with_bounded_error(x: 'Expr', eps: 'Expr' = None, m: int = 0,
+                             options: OPT_DICT = None) -> TMP_RES:
+    """
+    Evaluate *x* to within a bounded absolute error.
+
+    Parameters
+    ==========
+
+    x : Expr
+        The quantity to be evaluated.
+    eps : Expr, None, optional (default=None)
+        Positive real upper bound on the acceptable error.
+    m : int, optional (default=0)
+        If *eps* is None, then use 2**(-m) as the upper bound on the error.
+    options: OPT_DICT
+        As in the ``evalf`` function.
+
+    Returns
+    =======
+
+    A tuple ``(re, im, re_acc, im_acc)``, as returned by ``evalf``.
+
+    See Also
+    ========
+
+    evalf
+
+    """
+    if eps is not None:
+        if not eps.is_real or not eps > 0:
```

Comment:
Was wondering if there's an `is_Real` versus `is_real` consideration here. But `is_Real` doesn't seem to be a thing?

Reply:
Now I see that a `mypy` test is failing due to an `Expr` not having an `is_real` attribute. So, what's the right way to check whether an arbitrary `Expr` represents a real number?

Reply:
Every Expr has an `is_real` attribute so maybe mypy just needs to be told that. Something like this:
https://github.com/sympy/sympy/blob/d822fcba181155b85ff2b29fe525adbafb22b448/sympy/core/basic.py#L84-L85

In fact every Basic has `is_real` but I think that really it should only be defined for Expr (#19474).

Note though that `is_real` is a *semantic* check (e.g. `Symbol('x', real=True).is_real`) which is not what is wanted here. Here I think that what is wanted is for `eps` to be an explicit elementary number like `Float/Rational`.

Reply:
Okay, thanks. I've updated the type checking.